google-ads/src/main/java/com/google/ads/googleads/lib/utils/FieldMasks.java

```diff
@@ -119,6 +124,56 @@ private static void compare(
     }
   }
 
+  /**
+   * Gets the field value of a message from a field mask path.
+   *
+   * @param fieldMaskPath The field mask path.
+   * @param entity The entity to retrieve values from.
+   * @return the field Object.
+   */
+  public static <V> V getFieldValue(String fieldMaskPath, Message entity) {
+    Deque<String> fieldMaskParts =
+        new LinkedList<>(Arrays.asList(fieldMaskPath.trim().split("\\.")));
+    Message currentEntity = entity;
+    while (!fieldMaskParts.isEmpty()) {
+      String fieldName = fieldMaskParts.remove();
+      Preconditions.checkNotNull(
+          currentEntity, String.format("Cannot get field value. %s is null", fieldName));
+      Descriptor descriptor = currentEntity.getDescriptorForType();
+      FieldDescriptor childField = descriptor.findFieldByName(fieldName);
+
+      if (childField == null) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Cannot retrieve field value. A matching field was not found after"
+                    + " navigating %s up to %s",
+                fieldMaskPath, fieldName));
+      }
+      V childValue = (V) currentEntity.getField(childField);
```

Comment:
Will this cast fail on intermediate fields, since `V` is the type of the leaf field, not all the fields leading up to it?
Instead, you could just keep the casts you have on lines 154 and 174.

Reply:
This seems to work for all fields in the path and won't compile without it, so I don't think we need to keep as is.

Reply:
I'm not sure if you'll see exceptions if you add tests as I mentioned in my [other comment](https://github.com/googleads/google-ads-java/pull/387#discussion_r542666622). Type erasure is weird in Java, so it's possible it'll still work, but I'll leave this open as a reminder to check after you address the other comment.